src/com/hms/flexyosisoftconnector/Main.java

```diff
@@ -189,7 +189,7 @@ private static void setCertificatePath(String path) {
   private static void setHttpTimeouts() {
     SysControlBlock SCB;
     boolean needsSave = false;
-    final String HTTPS_TIMEOUT_S = "2";
+    final String HTTPS_TIMEOUT_S = "120";

```

Comment:
The https timeout value should be a parameter read in from the config file, and then passed to this function. Not hardcoded in the function.

Reply:
@it-hms do you have more context behind this change? Generally the config files are kept to a bare minimum and things that would be changed/adapted by a user. I really wouldn't expect this parameter to be changed by a customer enough to warrant it being included in the config file. Alex Hawk and I recently discussed adding functionality in the config files to override finals defined in the jar, but nothing has been implemented regarding that. If this is addressing a specific issue related to a customer usage are they creating their own jars or using the public release?

Reply:
@TomKimsey 
One customer has a strange setup. The OSI experts believe the backend database is under provisioned. This is because it is just a test/development OSI setup. We are finding valid responses after 80 seconds. 
I don't think we should hardcode 120 seconds just because of this once customer. 
I like your suggestion. There should be a sane default. Then the config file should overwrite the default, if the parameter is included in the file.

Reply:
Yeah, if that is the case that is the best solution. If the json key exists on the config, use that value, if not use the default.

Reply:
An optional configuration parameter makes the most sense to me as well

Reply:
@oliver-walker-hms 
I don't know the difference between HTTPC_SDTO and HTTPC_RDTO. It might make sense to have two different parameters, and two different functions. Also, if the values change, it should be logged.

Reply:
HTTPC_SDTO HTTP client operations GetHTTP or PutHTTP - SEND timeout
HTTPC_RDTO HTTP client operations GetHTTP or PutHTTP - READ timeout
I think the issue is only relevant to reads but it should be fine to set both. I can have the configuration file only affect reads and test that this works

Reply:
@oliver-walker-hms 
Wow, this doesn't make too much sense. Just make it one function with one parameter.

Reply:
yeah the docs are not very clear. will do